storage/remote/queue_manager.go

```diff
@@ -435,6 +455,63 @@ func NewQueueManager(
 	return t
 }
 
+// LastMarkedSegment implements wal.Marker. It returns the last read segment
+// from the queue-specific file. Requests the WAL watcher to start at the end
+// of the WAL if the marker file doesn't exist yet.
+func (t *QueueManager) LastMarkedSegment() (segment *int) {
+	dir := filepath.Join(t.walDir, "remote", t.client().Name())
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		level.Warn(t.logger).Log("msg", "marker segment does not exist")
+		return
+	} else if err != nil {
+		level.Error(t.logger).Log("msg", "could not access segment marker folder", "dir", dir, "err", err)
+		return
+	}
+
+	fname := filepath.Join(dir, "segment")
+	bb, err := ioutil.ReadFile(fname)
+	if os.IsNotExist(err) {
+		level.Warn(t.logger).Log("msg", "marker segment file does not exist")
+		return
+	} else if err != nil {
+		level.Error(t.logger).Log("msg", "could not access segment marker file", "file", fname, "err", err)
+		return
+	}
+
+	savedSegment, err := strconv.Atoi(string(bb))
+	if err != nil {
+		level.Error(t.logger).Log("msg", "could not read segment marker file", "file", fname, "err", err)
+		return
+	}
+	return &savedSegment
+}
+
+func (t *QueueManager) markSegment(segment int) {
+	dir := filepath.Join(t.walDir, "remote", t.client().Name())
+	if err := os.MkdirAll(dir, 0o770); err != nil {
```

Comment:
is `0o770` valid?

Reply:
Yeah, it's equivalent to passing 0770, if you're asking about the syntax. Though maybe I was quick to jump to removing non-owner/non-group permissions?

Reply:
This is the same syntax [used in the TSDB](https://github.com/prometheus/prometheus/blob/e239e3ee8b13b51b0f791a199813a14f74600a7e/tsdb/wal.go#L176). But also, since that's 0o777, I'll use the same here